anitya/tests/test_flask_api_v2.py

```diff
@@ -481,6 +483,22 @@ def test_valid_request(self):
         data = _read_json(output)
         self.assertEqual({"distribution": "Fedora", "name": "python-requests"}, data)
 
+        project = models.Project.query.filter_by(
+            name="requests", ecosystem_name="pypi"
+        ).one()
+
+        msg = {
+            "project": dict(project.__json__()),
+            "distro": {"name": "Fedora"},
+            "message": {
+                "agent": "user@fedoraproject.org",
+                "project": "requests",
+                "distro": "Fedora",
+                "new": "python-requests",
+            },
+        }
+        mock_publish.assert_called_with(topic="project.map.new", msg=msg)
```

Comment:
you use fedora-messaging `testing.mock_sends` for that if you prefer. See example here --> https://pagure.io/waiverdb/blob/master/f/tests/test_events.py#_31

Reply:
Thanks, I will change it.

Reply:
I tried to implement the `testing.mock_sends`, but I'm getting this in tox:
```
TypeError: test_valid_request() missing 1 required positional argument: 'mock_publish'
```

The method looks like this now:
```python
    def test_valid_request(self, mock_publish):
        """Assert packages can be created."""
        request_data = {
            "project_ecosystem": "pypi",
            "project_name": "requests",
            "package_name": "python-requests",
            "distribution": "Fedora",
        }
        project = models.Project.query.filter_by(
            name="requests", ecosystem_name="pypi"
        ).one()

        msg = {
            "project": dict(project.__json__()),
            "distro": {"name": "Fedora"},
            "message": {
                "agent": "user@fedoraproject.org",
                "project": "requests",
                "distro": "Fedora",
                "new": "python-requests",
            },
        }

        expected_msg = api.Message(
            topic="anitya.project.map.new",
            body=msg
        )

        with testing.mock_sends(expected_msg):
            output = self.app.post(
                "/api/v2/packages/", headers=self.auth, data=request_data
            )

        self.assertEqual(output.status_code, 201)
        data = _read_json(output)
        self.assertEqual({"distribution": "Fedora", "name": "python-requests"}, data)
```

Reply:
I will leave it like it is now, because of the change in `project` object that is done inside the post request.